### PR TITLE
fix udeps and add to pr-tests

### PR DESCRIPTION
**Pull request by leonardoalt**
This PR fixes udeps again and finally adds it to the pr-tests so it fails in a PR already.

### CODE REVIEW

File: asmopt/Cargo.toml [hunk 1]
```diff
@@ -9,7 +9,6 @@ repository.workspace = true
 [dependencies]
 powdr-ast.workspace = true
 powdr-analysis.workspace = true
-powdr-importer.workspace = true
```
Comment (leonardoalt) [inline asmopt/Cargo.toml:12]: this is the fix

File: .github/workflows/pr-tests.yml [hunk 2]
```diff
@@ -247,45 +247,66 @@ jobs:
...
+          name: Benchmarks
+          tool: "cargo"
+          output-file-path: output.txt
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          auto-push: true
+          alert-threshold: "120%"
+          comment-on-alert: true
+          summary-always: true
+
+  udeps:
```
Comment (leonardoalt) [inline .github/workflows/pr-tests.yml:293]: This was added now.

File: .github/workflows/pr-tests.yml [hunk 3]
```diff
@@ -23,220 +23,220 @@ jobs:
...
-          ~/.cargo/registry/index/
-          ~/.cargo/registry/cache/
-          ~/.cargo/git/db/
-          target/
-          Cargo.lock
-        key: ${{ runner.os }}-cargo-pr-tests
-    - name: Date of the restored cache
-      run: cat target/cache-build-date.txt
-      continue-on-error: true
+      - uses: actions/checkout@v4
```
Comment (leonardoalt) [inline .github/workflows/pr-tests.yml:26]: this was all fmt, no idea why

**Review by Schaeff — APPROVED** [review]

File: .github/workflows/pr-tests.yml (hunk 3 above)
Comment (Schaeff) [reply .github/workflows/pr-tests.yml:26]: you mean `cargo fmt` ? Would it mean it looks at yml files as well?

**Review by Schaeff — APPROVED** [review]

File: .github/workflows/pr-tests.yml [hunk 4]
```diff
@@ -247,45 +247,65 @@ jobs:
...
+          tool: "cargo"
+          output-file-path: output.txt
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          auto-push: true
+          alert-threshold: "120%"
+          comment-on-alert: true
+          summary-always: true
+
+  udeps:
+    needs: build
```
Comment (chriseth) [inline .github/workflows/pr-tests.yml:294]: Why does it need build?

**Review by chriseth — APPROVED** [review]

**Pull request merged**